frontend/model/contracts/group.js

```diff
@@ -261,38 +261,79 @@ function updateGroupStreaks ({ state, getters }) {
   }
 }
 
+const removeGroupChatroomProfile = (state, chatRoomID, member) => {
+  console.log('jjjj removeGroupChatroomProfile', JSON.parse(JSON.stringify({ state, chatRoomID, member })))
+  Vue.set(state.chatRooms[chatRoomID], 'users',
+    Object.fromEntries(
+      Object.entries(state.chatRooms[chatRoomID].users)
+        .map(([memberKey, profile]) => {
+          console.log('jjjj removeGroupChatroomProfile ii', {
+            memberKey, member, profile, active: (profile: any)?.status === PROFILE_STATUS.ACTIVE
+          })
+          if (memberKey === member && (profile: any)?.status === PROFILE_STATUS.ACTIVE) {
+            return [memberKey, { status: PROFILE_STATUS.REMOVED }]
+          }
+          return [memberKey, profile]
+        })
+    )
+  )
+}
+
 const leaveChatRoomAction = (state, { chatRoomID, member, leavingGroup }, meta) => {
   const sendingData = leavingGroup
     ? { member: member }
     : { member: member, username: meta.username }
 
+  if (state?.chatRooms?.[chatRoomID]?.users?.[member]?.status !== PROFILE_STATUS.REMOVED) {
+    return
+  }
+
+  const extraParams = {}
+
+  // When a group is being left, we want to also leave chatrooms,
+  // including private chatrooms. Since the user issuing the action
+  // may not be a member of the chatroom, we use the group's CSK
+  // unconditionally in this situation, which should be a key in the
+  // chatroom (either the CSK or the groupKey)
+  if (leavingGroup) {
+    const signingKeyId = sbp('chelonia/contract/currentKeyIdByName', state, 'csk', true)
+
+    // If we don't have a CSK, it is because we've already been removed.
+    // Proceeding would cause an error
+    if (!signingKeyId) {
+      return
+    }
+
+    extraParams.signingKeyId = signingKeyId
+    extraParams.innerSigningContractID = null
```

Comment:
Does `innerSigningContractID = null` need to be specified? If yes, could you add a comment explaining why?

Reply:
Yes, it needs to be specified or else it'll be set to the identity CSK.